src/opm/input/eclipse/Schedule/KeywordHandlers.cpp

```diff
@@ -766,19 +766,69 @@ File {} line {}.)", wname, location.keyword, location.filename, location.lineno)
+            const auto& pressure_item = record.getItem<ParserKeywords::GRUPNET::TERMINAL_PRESSURE>();
+            const int vfp_table = record.getItem<ParserKeywords::GRUPNET::VFP_TABLE>().get<int>(0);
 
-            auto new_group = this->snapshots.back().groups.get( groupName );
-            if (new_group.updateNetVFPTable(table))
-                this->snapshots.back().groups.update( std::move(new_group) );
+            for (const auto& group_name : group_names) {
+                const auto& group = this->snapshots.back().groups.get(group_name);
+                const auto& parent_name = group.parent();
+                if (!parent_name.empty())
+                {
+                    const std::string& downtree_node = group_name;
+                    const std::string& uptree_node = parent_name;
+                    if (vfp_table == 0) {
+                        // If vfp table is defaulted (or set to 0) then the group is not part of the network
+                        if (network.has_node(downtree_node) && network.has_node(uptree_node)) 
+                            // If it was part of the network then drop it
+                            network.drop_branch(uptree_node, downtree_node);
+                    } else {
+                         const auto alq_eq = Network::Branch::AlqEqfromString(record.getItem<ParserKeywords::GRUPNET::ALQ_SURFACE_DENSITY>().get<std::string>(0));
+                         if (alq_eq == Network::Branch::AlqEQ::ALQ_INPUT) {
+                             const double alq_value = record.getItem<ParserKeywords::GRUPNET::ALQ>().get<double>(0);
+                             network.add_branch(Network::Branch(downtree_node, uptree_node, vfp_table, alq_value));
+                        } else {
+                             network.add_branch(Network::Branch(downtree_node, uptree_node, vfp_table, alq_eq));
+                        }
+                    }
+                }
+                bool is_valid_node = false;
+                Network::Node node { group_name };
+                const bool is_terminal_node = pressure_item.hasValue(0) && (pressure_item.get<double>(0) >= 0);
+                if (is_terminal_node) {
+                    if (vfp_table == 0) {
+                        is_valid_node = true;
+                        node.terminal_pressure(pressure_item.getSIDouble(0));
+                    } else {
+                         std::string msg = fmt::format("The group {} is a terminal node of the network and should not have a vfp table assigned to it.", group_name);
+                         throw OpmInputError(msg, handlerContext.keyword.location());
+                    }
+                } else {
+                     if (vfp_table != 0){
+                         // Only non-terminal nodes with non-default vfp table value can be part of the network
+                         is_valid_node = true;
+                     } else {
+                          if(network.has_node(node.name())) {
```

Comment:
Should not we remove the branch here instead of throw?

Reply:
I think so too. In the new structure the branch is removed.